apps/microsoft-teams/frontend/src/components/config/ChannelSelection/ChannelSelection.tsx

```diff
@@ -16,18 +16,31 @@ interface Props {
 
 const ChannelSelection = (props: Props) => {
   const { notification, handleNotificationEdit } = props;
-  const { channels } = useContext(ChannelContext);
+  const { channels, loading, error } = useContext(ChannelContext);
+  const [areChannelsLoading, setAreChannelsLoading] = useState<boolean>(false);
+  const [addButtonClicked, setAddButtonClicked] = useState<boolean>(false);
+
+  useEffect(() => {
+    // ensure the loading state updates once it is done loading
+    if (addButtonClicked) setAreChannelsLoading(loading);
```

Comment:
Yeah so I think the original intention here is to only handle this as an edge case of loading, as most likely and ideally the user will never see the loading state itself because everything will be loaded. This acts more as a safeguard to this edge case that the user beats our app action while configuring things. But that said, I also implemented it the way you are describing to see how it looks and it is also a fine user experience imo, so am happy to do either way, and perhaps is more transparent with the user that these things are loading! I will check with Emily to confirm 😄

Reply:
Same here, based on Emily's feedback, let's just keep it as is!